### ISSUE TITLE
Intermittent Resource of class not found: io.katharsis.response.LinkageContainer

### ISSUE THREAD

**Issue by csetera** [katharsis-project/katharsis-core#131]
This is driving me crazy.... I'm building out a test harness for testing with and developing for Katharsis.  The test harness uses an embedded Jetty instance configured to use the katharsis-servlet support to configure Katharsis.  For resource repositories, things are working fine.  However, for relationship repositories, it seems that *most* of the time I see the attached errors, but not every time.  In fact, many times the act of trying to debug the issue results in things working fine.  Here are the stack traces:

```java
io.katharsis.invoker.KatharsisInvokerException: com.fasterxml.jackson.databind.JsonMappingException: Resource of class not found: io.katharsis.response.LinkageContainer
	at io.katharsis.invoker.KatharsisInvoker.invoke(KatharsisInvoker.java:73)
	at io.katharsis.servlet.AbstractKatharsisFilter.doFilter(AbstractKatharsisFilter.java:86)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1668)
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:581)
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1158)
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:511)
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1090)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:119)
	at org.eclipse.jetty.server.Server.handle(Server.java:517)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:306)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:242)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:261)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:95)
	at org.eclipse.jetty.io.SelectChannelEndPoint$2.run(SelectChannelEndPoint.java:75)
	at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.produceAndRun(ExecuteProduceConsume.java:213)
	at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.run(ExecuteProduceConsume.java:147)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:654)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$3.run(QueuedThreadPool.java:572)
	at java.lang.Thread.run(Thread.java:745)
Caused by: com.fasterxml.jackson.databind.JsonMappingException: Resource of class not found: io.katharsis.response.LinkageContainer
	at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider.serializeValue(DefaultSerializerProvider.java:140)
	at com.fasterxml.jackson.databind.ObjectMapper._configAndWriteValue(ObjectMapper.java:3385)
	at com.fasterxml.jackson.databind.ObjectMapper.writeValue(ObjectMapper.java:2745)
	at io.katharsis.invoker.KatharsisInvoker.dispatchRequest(KatharsisInvoker.java:120)
	at io.katharsis.invoker.KatharsisInvoker.invoke(KatharsisInvoker.java:71)
	... 19 more
Caused by: io.katharsis.resource.exception.init.ResourceNotFoundInitializationException: Resource of class not found: io.katharsis.response.LinkageContainer
	at io.katharsis.resource.registry.ResourceRegistry.getEntry(ResourceRegistry.java:43)
	at io.katharsis.jackson.serializer.IncludedRelationshipExtractor.getRelationshipFields(IncludedRelationshipExtractor.java:152)
	at io.katharsis.jackson.serializer.IncludedRelationshipExtractor.getIncludedByDefaultResources(IncludedRelationshipExtractor.java:61)
	at io.katharsis.jackson.serializer.IncludedRelationshipExtractor.extractDefaultIncludedFields(IncludedRelationshipExtractor.java:47)
	at io.katharsis.jackson.serializer.IncludedRelationshipExtractor.extractIncludedResources(IncludedRelationshipExtractor.java:35)
	at io.katharsis.jackson.serializer.BaseResponseSerializer.serializeResourceCollection(BaseResponseSerializer.java:81)
	at io.katharsis.jackson.serializer.BaseResponseSerializer.serialize(BaseResponseSerializer.java:44)
	at io.katharsis.jackson.serializer.BaseResponseSerializer.serialize(BaseResponseSerializer.java:21)
	at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider.serializeValue(DefaultSerializerProvider.java:129)
	... 23 more
```

The fact that it works sometimes makes me believe that this isn't directly my code.  The fact that it is intermittent makes it seem to be either timing or asynchronous, however I've been unable to find anything to nail down the issue.  I could really use some suggestions.

**Comment by Sandmania** [comment]
I've encountered this same error quite a few times as well, with pretty much the same weird behaviour: when I start debugging it (say, attach a remote debugger), the problem disappears. Still haven't found a way to reliable reproduce this.

**Comment by meshuga** [comment]
Could you tell if the problem occurs when doing something different that what's described in https://github.com/katharsis-project/katharsis-core/issues/119 ?

**Comment by csetera** [comment]
This looks like the same problem, although what I still don't understand is how this can be inconsistent and intermittent?  I had hoped to look at this more last night, but did not get a chance.  I had started to wonder if this was somehow tied to classloaders.  

It appears that #119 should be available on the master branch?  I can always build a local copy and try it out... although I did notice there were some breaking changes to the API which will require some changes on my end.  Is there an ETA for the 2.0 release?
